### PR TITLE
Added cash-js.d.ts File

### PR DESCRIPTION
**Pull request by avishwakarma**
case 1. Add a new type definition.
- checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url https://github.com/kenwheeler/cash .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by vvakame** [comment]
https://www.npmjs.com/package/cash-dom

**Comment by RyanCavanaugh** [comment]
Is the module `cash-js`, `Cash`, or `cash-dom` ? Please rename the folder, module declaration, and files to all match whatever it is on NPM

**Comment by akvlko** [comment]
Its a definition file for CashJs https://github.com/kenwheeler/cash 
Do you want me change name from `cash-js` to `cash`? As the github repo name is `cash`

**Comment by akvlko** [comment]
Made the changes suggested by @RyanCavanaugh :)

File: cash/cash.d.ts [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+  val(value?: string): Cash;
+
+  /**
+   * width
+   * Returns the width of the element.
+   */
+  width(): number;
+}
+
+declare module "Cash" {
```
Comment (RyanCavanaugh) [inline cash/cash.d.ts]: This should be lower-case (it should match the containing folder name, basically)

**Comment by akvlko** [comment]
Changed the module name as suggested by @RyanCavanaugh

File: cash/cash.d.ts [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+   * Triggers supplied event on elements in collection. Data can be passed along as the second parameter.
+   */
+  trigger(eventName: string, data?: any): Cash;
+
+  /**
+   * val
+   * Returns an inputs value. If value is supplied, sets all inputs in collection's value to the value argument.
+   */
+  val(): any;
+  val(value?: string): Cash;
```
Comment (andy-ms) [inline cash/cash.d.ts]: `value` here should be non-optional, since the first overload will cover that case anyway.

File: cash/cash.d.ts [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+  isNumeric(n: any): boolean;
+
+  /**
+   * isString
+   * Check if the argument is a string.
+   * @type method
+   * @argument str any
+   * @return boolean
+   */
+  isString(str: any): boolean;
```
Comment (andy-ms) [inline cash/cash.d.ts]: This could be `: str is string`.

File: cash/cash.d.ts (hunk 2 above)
Comment (andy-ms) [inline cash/cash.d.ts]: `value` here should be non-optional, since the first overload will handle that case.

File: cash/cash.d.ts (hunk 3 above)
Comment (andy-ms) [inline cash/cash.d.ts]: This could return `str is string`.

**Comment by mhegazy** [comment]
@akvlko we would appreciate it if you can get the comments above addressed in a new PR.

**Pull request merged**